Mage.Sets/src/mage/cards/e/EvolvingAdaptive.java

```diff
@@ -0,0 +1,99 @@
+package mage.cards.e;
+
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.*;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.common.CountersSourceCount;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.continuous.BoostSourceEffect;
+import mage.abilities.effects.common.counter.AddCountersSourceEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.counters.CounterType;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+
+import java.util.UUID;
+
+public class EvolvingAdaptive extends CardImpl {
+
+    private static final FilterPermanent filter = new FilterCreaturePermanent("another creature");
+
+    static {
+        filter.add(AnotherPredicate.instance);
+    }
+
+    private static final DynamicValue oilCounters = new CountersSourceCount(CounterType.OIL);
+
+    public EvolvingAdaptive(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{G}");
+        this.addSubType(SubType.PHYREXIAN);
+        this.addSubType(SubType.WARRIOR);
+        this.power = new MageInt(0);
+        this.toughness = new MageInt(0);
+
+        //Evolving Adaptive enters the battlefield with an oil counter on it.
+        this.addAbility(new EntersBattlefieldAbility(
+                new AddCountersSourceEffect(CounterType.OIL.createInstance()),
+                "with an oil counter on it"
+        ));
+
+        //Evolving Adaptive gets +1/+1 for each oil counter on it.
+        this.addAbility(new SimpleStaticAbility(new BoostSourceEffect(oilCounters, oilCounters, Duration.WhileOnBattlefield)));
+
+        //Whenever another creature enters the battlefield under your control, if that creature has greater power or
+        //toughness than Evolving Adaptive, put an oil counter on Evolving Adaptive.
+        this.addAbility(new EntersBattlefieldControlledTriggeredAbility(new EvolvingAdaptiveEffect(), filter));
```

Comment:
This needs to either use its own custom triggered ability with an overridden `checkTrigger` as in the case with [Ichorplate Golem](https://github.com/magefree/mage/blob/5c48d05898402eaa97064c721886a1a9ab23c2e5/Mage.Sets/src/mage/cards/i/IchorplateGolem.java) or it needs to use a `ConditionalInterveningIfTriggeredAbility` as in the case with [Liberator, Urza's Battlethopter](https://github.com/magefree/mage/blob/611d6d6a45e00231425b37b097698682d2b15bb5/Mage.Sets/src/mage/cards/l/LiberatorUrzasBattlethopter.java)

Reply:
I did implement a custom TriggeredAbility.